### ISSUE TITLE
HTML does not open in Viewer if Rmd is outside current working directory and has parentheses in the filename, and Knit directory is set to Current working directory.

### ISSUE THREAD

**Issue by jonovik** [rstudio/rstudio#13735]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2023.09.0 Build 463
    OS Version      : Windows 10 Version 21H2 OS Build 19044.3448
    R Version       : 4.3.1

### Steps to reproduce the problem

1. Open RStudio.
2. File > New file > R markdown > accept all defaults
3. Save this file somewhere other than the current working directory and call it `test (1).Rmd`. I used `c:\test\test (1).Rmd`.
4. Using the dropdown on the Knit button, set Knit directory > Current working directory.
5. Press the Knit button.
6. The Viewer pane remains blank.

### Describe the problem in detail

The Render pane shows the following output. Note how spaces and parentheses are replaced with hyphens in the md and html output filenames. The problem does not manifest with just spaces in the filename; `a test.Rmd` is okay.

```
  |.................                                          |  29% (setup)   

processing file: test--1-.Rmd
                                                                                                     
output file: C:/git/teach/stin100/test--1-.knit.md

"C:/Program Files/RStudio/resources/app/bin/quarto/bin/tools/pandoc" +RTS -K512m -RTS "C:/git/teach/stin100/test--1-.knit.md" --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash --output pandoc30c044025c9e.html --lua-filter "C:\r-library\rmarkdown\rmarkdown\lua\pagebreak.lua" --lua-filter "C:\r-library\rmarkdown\rmarkdown\lua\latex-div.lua" --embed-resources --standalone --variable bs3=TRUE --section-divs --template "C:\r-library\rmarkdown\rmd\h\default.html" --no-highlight --variable highlightjs=1 --variable theme=bootstrap --mathjax --variable "mathjax-url=https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" --include-in-header "C:\NMBU\TEMP\RtmpYJ12tK\rmarkdown-str30c056174e7.html" 

Output created: test--1-.html
```

The problem goes away if the filename is `c:\temp\test2.Rmd` without parentheses. Its report correctly opens in the Viewer pane.

The problem also goes away if I switch the Knit directory to Document directory and re-knit `c:\temp\test (1).Rmd`. The report opens correctly.

### Describe the behavior you expected

The generated HTML report should open in the Viewer pane like it already does when Knit directory is "document directory" or the filename does not contain parentheses.

My best guess is that the filename mangling (replacing spaces and parentheses with hyphens) happens slightly differently when knitting and when RStudio tries to open the html file in Viewer.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.posit.co/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@jonovik Thank you for raising the issue! I can also reproduce it in the above environment, as well as with RStudio Desktop 2022.07.2+576 and 2023.12.0-daily+165 on MacOS 14. Occurs with both files in Viewer Pane and separate window, and with both current and project directories. Works fine in document directory.

**Comment by jonovik** [comment]
This seems to be fixed in 2023.12.1 build 402 .

```
RStudio 2023.12.1+402 "Ocean Storm" Release (4da58325ffcff29d157d9264087d4b1ab27f7204, 2024-01-28) for windows
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2023.12.1+402 Chrome/116.0.5845.190 Electron/26.2.4 Safari/537.36
```
